keymaster/ipc/optee_keymaster_ipc.cpp

```diff
@@ -251,7 +251,15 @@ keymaster_error_t optee_keymaster_call(uint32_t cmd,
     } else if (rsp->error != KM_ERROR_OK) {
 	ALOGE("Response of size %d contained error code %d\n", (int)rsp_size,
 	      (int)rsp->error);
-	return rsp->error;
+    } else if (res != KM_ERROR_OK) {
+	/*
+	 * rsp->error is KM_ERROR_OK but res isn't? The only time this happens
+	 * is in the case of param_types != exp_param_types in
+	 * TA_InvokeCommandEntryPoint() so res can't be serialized into rsp.
+	 */
+	ALOGE("Response of size %d contained error code %d\n", (int)rsp_size,
+	      (int)res);
+	return (keymaster_error_t)res;
```

Comment:
res looks like TEE_ERROR. Wouldn't typecasting it to keymaster_error_t result in wrong errorcode

Reply:
The `TEE_Result` from `TA_InvokeCommandEntryPoint()` is actually a `keymaster_error_t` since all the functions called by `TA_InvokeCommandEntryPoint()` return a `keymaster_error_t` so here we cast it to match.